sdk/identity/azure-identity/azure/identity/aio/_credentials/on_behalf_of.py

```diff
@@ -0,0 +1,87 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import logging
+from typing import TYPE_CHECKING
+
+from azure.core.exceptions import ClientAuthenticationError
+
+from .._internal import AadClient
+from .._internal.decorators import log_get_token_async
+from ..._internal import validate_tenant_id
+
+if TYPE_CHECKING:
+    from typing import Any
+    from azure.core.credentials import AccessToken
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OnBehalfOfCredential:
```

Comment:
Do we want to inherit from abc.ABC?

Reply:
Sorry. I meant AsyncContextManager. I need some coffee. :)

Reply:
Right you are. I needed coffee when I wrote this 💤